Marlin/src/lcd/dogm/status_screen_DOGM.cpp

```diff
@@ -508,6 +515,8 @@ void MarlinUI::draw_status_screen() {
 
   const bool show_e_total = TERN0(LCD_SHOW_E_TOTAL, printingIsActive());
 
+  static u8g_uint_t progress_bar_solid_width = 0;
+
   // At the first page, generate new display values
   if (first_page) {
```

Comment:
Note that with the regular DOGM display there are several calls per update to redraw the screen. This `first_page` flag indicates that we're drawing the first "stripe" in the screen update. Any expensive calculations or string composition is done here so that they only happen once per screen update. To optimize the actual draw operations, we use `if (...) { ... }` with `PAGE_OVER`, `PAGE_UNDER`, and `PAGE_CONTAINS` to check whether the thing we're drawing is within the current stripe, and if not then drawing is skipped.